### PR TITLE
fix: tup-615 inconsistent header spacing - left & vertical

### PR DESCRIPTION
**Pull request by wesleyboar**
## Overview

Make header spacing consistent.

## Related

- [TUP-615](https://jira.tacc.utexas.edu/browse/TUP-615)

## Changes

- #338
- #340
- #341
- #342
- [fix: project ticket grid too much middle space (`f027640`)](https://github.com/TACC/tup-ui/pull/339/commits/f027640b0a3991ba770f8758450b6e0d02408185)

## Testing

1. Open all sections.
2. Verify all header text has the same space before and above.
3. Verify, in Dashboard section, on large window (or with small text):
    - no excess space above "Dashboard" header text
    - no excess space between "Active Projects" and "My Tickets" table

## UI

| | | |
| - | - | - |
| dashboard | proj alloc | project
tickets | system | account |

| dashboard layout |
| - |
| dashboard - full |

The cut-off System Status "Running Jobs" is caused by #312, not this.

### CODE REVIEW

**Review by jarosenb — APPROVED** [review]
Looks good in dev!

**Review by jarosenb — CHANGES_REQUESTED** [review]
We have an issue if the projects listing is large. It can crowd out the tickets and make them completely inaccessible: 
image
Maybe add a max-height on the project table?

**Comment by wesleyboar** [comment]
I tested two fixes to [the problem you found](https://github.com/TACC/tup-ui/pull/339#pullrequestreview-1723624476).

**On the Grid:**
```css
.project-ticket-grid {
    ...
    grid-template-rows: minmax(auto, 200px) 1fr; /* instead of auto 1fr */
    ...
}
```

**On the Grid Item:**
```css
.project-ticket-grid:nth-child(1) {
    ...
    max-height: 200px;
}
```

**Comment by wesleyboar** [comment]
New fix for narrow and wide screen (2689615). Slimmer and clearer I think.

https://github.com/TACC/tup-ui/assets/62723358/52a38bbb-44bf-4e47-bb27-c9fed4b5393d

**Review by R-Tomas-Gonzalez — COMMENTED** [review]
Question on headers and consistency. I noticed that some headers have nested headers, and some do not. The headers that are **not** nested, seem to have padding-right, due to being within articles. The headers that **are** nested, span the whole width of the page. Right padding may not matter. But thought I'd mention it.
 Here are the examples:
 
Screenshot 2023-11-14 at 11 20 06 AM
Screenshot 2023-11-14 at 11 19 10 AM
Screenshot 2023-11-14 at 11 19 50 AM
Screenshot 2023-11-14 at 11 19 33 AM

**Comment by R-Tomas-Gonzalez** [comment]
*Note:* 

- https://github.com/TACC/tup-ui/pull/338
- https://github.com/TACC/tup-ui/pull/340
- https://github.com/TACC/tup-ui/pull/341 _(except for comment above)_
- https://github.com/TACC/tup-ui/pull/342
- [fix: project ticket grid too much middle space (f027640)](https://github.com/TACC/tup-ui/pull/339/commits/f027640b0a3991ba770f8758450b6e0d02408185)

All look good to me

**Comment by wesleyboar** [comment]
Hm, thanks @R-Tomas-Gonzalez for https://github.com/TACC/tup-ui/pull/339#pullrequestreview-1730311908.

I did not pay attention to the right side of the headers…

I'll check this, and how #335 interacts with it.

**Comment by wesleyboar** [comment]
@R-Tomas-Gonzalez What do you think consistent space on the right being handled in a new ticket instead of this one?

<details><summary>My reasoning…</summary>

- original ticket was focused on left and vertical
- this PR seems to be a valid fix for left and vertical
- i think it may be easier to tackle right-hand side in isolation
- i think it will be easier to tackle right-hand side after merges of existing PRs
</details>

**Comment by R-Tomas-Gonzalez** [comment]
> @R-Tomas-Gonzalez What do you think consistent space on the right being handled in a new ticket instead of this one?
> 
> My reasoning…
> * original ticket was focused on left and vertical
> * this PR seems to be a valid fix for left and vertical
> * i think it may be easier to tackle right-hand side in isolation
> * i think it will be easier to tackle right-hand side after merges of existing PRs

I agree. 👍 :)

**Comment by R-Tomas-Gonzalez** [comment]
Will approve when conflicts are resolved.

**Comment by wesleyboar** [comment]
New ticket: [TUP-651](https://jira.tacc.utexas.edu/browse/TUP-651)
Consistent Spacing on Right for Section Headers

**Review by R-Tomas-Gonzalez — APPROVED** [review]

**Review by jarosenb — APPROVED** [review]
Approving now that long project lists no longer overflow the column.

**Comment by wesleyboar** [comment]
> **Warning**
> Do not merge. Diff has accidental merge of CSS migration via 0e84c10.

**Comment by wesleyboar** [comment]
> **Important**
> Safe to merge. Problem resolved. I reverted 0e222ee via f6b7bd5. Retested. Re-reviewed diff.

File: apps/tup-ui/src/pages/Dashboard/Dashboard.module.css
Comment (wesleyboar) [inline apps/tup-ui/src/pages/Dashboard/Dashboard.module.css:32]: This performs #369 **again**, but is tested at 100% zoom. ✔️ #369 was tested at 90% zoom. 🫤

**Pull request merged**
